aydin/nn/models/utils/masking/random_masked_dataset.py

```diff
@@ -0,0 +1,61 @@
+import numpy
+import torch
+from torch.utils.data import Dataset
+
+
+class RandomMaskedDataset(Dataset):
+    def __init__(
+        self,
+        image,
+        nb_masks=4,
+        pixel_masking_probability: float = 0.3,
+    ):
+        """
+
+        Parameters
+        ----------
+        image
+        """
+
+        self.image = torch.tensor(image)
+        self.nb_masks = nb_masks
+        self.p = pixel_masking_probability
+
+    def __len__(self):
+        return self.nb_masks
+
+    def get_mask(self, i):
+        shape = self.image.shape
+
+        mask = torch.rand(shape)
+        mask[mask > self.p] = 1
+        mask[mask <= self.p] = 0
+        mask = mask.bool()
+
+        return mask
+
+    def interpolate_mask(self, tensor, mask, mask_inv):
+        device = tensor.device
+
+        mask = mask.to(device)
+
+        kernel = numpy.array([[0.5, 1.0, 0.5], [1.0, 0.0, 1.0], (0.5, 1.0, 0.5)])
+        kernel = kernel[numpy.newaxis, numpy.newaxis, :, :]
+        kernel = torch.Tensor(kernel).to(device)
+        kernel = kernel / kernel.sum()
+
+        filtered_tensor = torch.nn.functional.conv2d(
+            tensor, kernel, stride=1, padding=1
+        )
+
+        return filtered_tensor * mask + tensor * mask_inv
```

Comment:
You only explicitly send `mask` and `kernel` to `device`. Are you sure `tensor` and `mask_inv` are already in `device` too? If not, I'm afraid that errors might be raised.

Reply:
great catch! thanks